pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV3.java

```diff
@@ -175,8 +175,11 @@ public DataTableImplV3(ByteBuffer byteBuffer)
   }
 
   @Override
-  public void addException(ProcessingException processingException) {
-    _errCodeToExceptionMap.put(processingException.getErrorCode(), processingException.getMessage());
+  public void addException(Exception exception) {
+    if (exception instanceof ProcessingException) {
```

Comment:
any chance we add other exceptions here?
Shall we add a log here if the exception is not ProcessingException?